releasenotes/notes/refactor-reconstruct-45e00c3df1bdd4ff.yaml

```
@@ -0,0 +1,4 @@
+---
+upgrade:
+  - |
+    :func:`.reconstruct_expectation_values` now takes a :class:`~qiskit.primitives.SamplerResult` instance or a dictionary mapping partition labels to :class:`~qiskit.primitives.SamplerResult` instances as input to the first positional arg, ``results``. ``results`` replaces the old ``quasi_dists`` positional arg. The :class:`~qiskit.primitives.SamplerResult` instances are expected to contain the number of QPD bits used in each circuit input to the Sampler. This should be specified in the ``num_qpd_bits`` field of the experiment result metadata.
```

For some reason I prefer not abbreviating argument in the release notes.  This change seems to be consistent with the way most formal documentation is written.

While suggesting this change, though, I realized that these aren't _only_ positional arguments because there is nothing stopping them by being specified by keyword instead.  So I rephrased to make it clear that we are talking about the argument in the first position, not to be confused with https://peps.python.org/pep-0570/.
```suggestion
    :func:`.reconstruct_expectation_values` now takes, as its first argument, a :class:`~qiskit.primitives.SamplerResult` instance or a dictionary mapping partition labels to :class:`~qiskit.primitives.SamplerResult`\ s.  This new ``results`` argument replaces the old ``quasi_dists`` argument. The :class:`~qiskit.primitives.SamplerResult` instances are expected to contain the number of QPD bits used in each circuit input to the Sampler. This should be specified in the ``num_qpd_bits`` field of the experiment result metadata.
```